R/step-call.R

```diff
@@ -231,3 +231,43 @@ unique.dtplyr_step <- function(x, incomparables = FALSE, ...) {
   }
   distinct(x)
 }
+
+# tidyr verbs -------------------------------------------------------------
+#' Drop rows containing missing values
+#'
+#' @description
+#' This is a method for the tidyr `drop_na()` generic. It is translated to
+#' `data.table::na.omit()`
+#'
+#' @param data A [lazy_dt()].
+#' @inheritParams tidyr::drop_na
+#' @examples
+#' library(dplyr)
+#' library(tidyr)
+#'
+#' dt <- lazy_dt(tibble(x = c(1, 2, NA), y = c("a", NA, "b")))
+#' dt %>% drop_na()
+#' dt %>% drop_na(x)
+#'
+#' vars <- "y"
+#' dt %>% drop_na(x, any_of(vars))
+# exported onLoad
+drop_na.dtplyr_step <- function(data, ...) {
+  sim_data <- simulate_vars(data)
+  locs <- names(tidyselect::eval_select(expr(c(...)), sim_data))
+
+  args <- list()
+  if (length(locs) > 0) {
+    args$cols <- locs
+  }
+
+  step_call(data, "na.omit", args = args)
+}
+
+# exported onLoad
+drop_na.data.table <- function(data, ...) {
+  data <- lazy_dt(data)
+  drop_na(data, ...)
+}
+
+globalVariables("drop_na")
```

Comment:
You can instead add to `dt_funs` in `tidyeval.R`

Reply:
Isn't `dt_funs` just for data.table functions?

Reply:
Oh sorry, I totally misinterpreted the problem. You should use `tidyr::drop_na()` in the `drop_na` method.